libs/native-types/src/mssql.rs

```diff
@@ -1,56 +1,564 @@
+            MsSqlType::Binary(_) => 1,
+            MsSqlType::VarBinary(_) => 1,
+            MsSqlType::Image => 0,
+            MsSqlType::Xml => 0,
+        }
+    }
+
+    /// The name of the type without parameters.
+    pub fn kind(&self) -> &'static str {
+        match self {
+            MsSqlType::TinyInt => "TinyInt",
+            MsSqlType::SmallInt => "SmallInt",
+            MsSqlType::Int => "Int",
+            MsSqlType::BigInt => "BigInt",
+            MsSqlType::Decimal(_) => "Decimal",
+            MsSqlType::Numeric(_) => "Numeric",
+            MsSqlType::Money => "Money",
+            MsSqlType::SmallMoney => "SmallMoney",
+            MsSqlType::Bit => "Bit",
+            MsSqlType::Float(_) => "Float",
+            MsSqlType::Real => "Real",
+            MsSqlType::Date => "Date",
+            MsSqlType::Time => "Time",
+            MsSqlType::DateTime => "DateTime",
+            MsSqlType::DateTime2 => "DateTime2",
+            MsSqlType::DateTimeOffset => "DateTimeOffset",
+            MsSqlType::SmallDateTime => "SmallDateTime",
+            MsSqlType::Char(_) => "Char",
+            MsSqlType::NChar(_) => "NChar",
+            MsSqlType::VarChar(_) => "VarChar",
+            MsSqlType::Text => "Text",
+            MsSqlType::NVarChar(_) => "NVarChar",
+            MsSqlType::NText => "NText",
+            MsSqlType::Binary(_) => "Binary",
+            MsSqlType::VarBinary(_) => "VarBinary",
+            MsSqlType::Image => "Image",
+            MsSqlType::Xml => "Xml",
+        }
+    }
+
+    /// The type parameters, if any.
+    pub fn parameters(self) -> Vec<TypeParameter> {
+        match self {
+            MsSqlType::Decimal(Some((p, s))) => vec![TypeParameter::from(p), TypeParameter::from(s)],
+            MsSqlType::Numeric(Some((p, s))) => vec![TypeParameter::from(p), TypeParameter::from(s)],
+            MsSqlType::Float(Some(l)) => vec![TypeParameter::from(l)],
+            MsSqlType::Char(Some(l)) => vec![TypeParameter::from(l)],
+            MsSqlType::NChar(Some(l)) => vec![TypeParameter::from(l)],
+            MsSqlType::VarChar(Some(l)) => vec![l],
+            MsSqlType::NVarChar(Some(l)) => vec![l],
+            MsSqlType::VarBinary(Some(l)) => vec![l],
+            MsSqlType::Binary(Some(l)) => vec![TypeParameter::from(l)],
+            _ => vec![],
+        }
+    }
+}
+
+impl fmt::Display for MsSqlType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.kind())?;
+
+        let parameters = self.parameters();
+        let length = parameters.len();
+
+        if parameters.len() > 0 {
+            write!(f, "(")?;
+            for (i, p) in parameters.into_iter().enumerate() {
+                write!(f, "{}", p)?;
+
+                if i < length - 1 {
+                    write!(f, ",")?;
+                }
+            }
+            write!(f, ")")?;
+        }
+
+        Ok(())
+    }
+}
+
+impl FromStr for MsSqlType {
```

Comment:
discussed: add small comment to say it's made to handle two cases: native types stringified and parsed from the datamodel parser native types (PSL), and the types introspected from the database in sql-schema-describer